docs/GET_STARTED.md

```diff
@@ -1,6 +1,98 @@
+Whereas those in docs/examples/rx, use the functional approach of rxjs (Reactive Extensions Library) and treat api results as observable streams.
 
-## Test in local
+Here are the preparatory steps before you create your first project to interact with a CENNZnet node and consequently the CENNZnet blockchain network:
+1. Install or update NPM and NodeJs on your development platform. You should be good to go if you just use the latest versions. At the time of writing this guide we were using Node 14.5.0 and NPM 6.14.5.   
+2. Install yarn. yarn is our favorite package manager. However this is not a must if you know how to use any other package manager including NPM to produce the same instructions.
+3. Create a new NodeJs project or or use one of the examples to start with.
+
+NodeJs projects have a file called package.json in which the dependent libraries and their right versions are specified. The simplest project that you create would depend on "@cennznet/api". 
+You are here browsing the source code of this library. 
+The library as a package is published at: https://www.npmjs.com/package/@cennznet/api which you would need to check to figure out which version of the library is the right one for you given the version of the cennznet node that you want to connect to. 
+cennznet/api is internally dependant on polkadot. 
+At the moment a breaking change in the recent versions of polkadot has prevented us from updating with the recent versions. 
+That's why in package.json there should be a resolution set so we can make sure the right version of polkadot packages are fetched. 
+Therefore you need to have the following lines in your package.json:
+```json
+"dependencies": {
+  "@cennznet/api": "^1.0.3"
+},
+"resolutions": {
+  "@polkadot/types": "1.2.1",
+  "@polkadot/metadata": "1.2.1",
+  "@polkadot/api": "1.2.1",
+  "@polkadot/api-derive": "1.2.1",
+  "@polkadot/rpc-core": "1.2.1"
+}
+``` 
+The next step for your project is that you import @cennznet/api in your js code. The best way to do that is through a line such as:
+```js
+const { Api } = require('@cennznet/api');
+```
+When you have the api, you can easily connect to any cennznet node by calling "connect". 
+The underlying mechanism to connect to a node is through websockets. 
+So what you need to have for this call is the websocket address of your node. 
+The cennznet's main network is residing at `wss://cennznet.unfrastructure.io/public/ws`. 
+If you use this address to connect to our main network which is called Azalea, you automatically get connected to one of the cennznet's validator/full nodes. 
+However you may want to run your own local node and connect to that for testing. To do so you can checkout the docker-compose script that we have provided in this project and also mentioned it later in this document. 
+Alternatively you can have a look at a https://github.com/cennznet/cennznet/wiki/Running-a-Full-Node. 
```

Comment:
may be add a link for https://github.com/cennznet/cennznet/wiki/Running-a-Full-Node

Reply:
done